### DISCUSSION TITLE
configuration for LXC Container

### DISCUSSION THREAD

**Discussion by gamer123** [angristan/wireguard-install#421] [Help]
Hi 

thanks for provide that great script. 
I run it actually on ubuntu 20.04.1 LTS on a rasperry pi 4 4GB and it is working impressive. 

But to save money(less powerconsumption) i would like to run the script on my QNAP TS 231P Server ARM 32 bit AL212 CPU.
in/on a LXC container Ubuntu 18.04.

You Wrote 
"but the kernel module has to be installed on the host,                                                                                                             
the container has to be run with some specific parameters                                                                                                          
and only the tools need to be installed in the container.                                                                                                          
Welcome to the WireGuard installer!  "

I only have at the end this fault:
Job for wg-quick@wg0.service failed because the control process exited with error code.                                                                            
See "systemctl status wg-quick@wg0.service" and "journalctl -xe" for details.                                                                                      
Created symlink /etc/systemd/system/multi-user.target.wants/wg-quick@wg0.service → /lib/systemd/system/wg-quick@.service.  

-->do i need a special kernel for ?

QNAP TS 231P Server ARM 32 bit AL212 CPU
AnnapurnaLabs, an Amazon company Alpine AL212, 2-core, 1.7GHz

-->wich are the "specific parameters" parameters ? i couldn´t find something in the www. 

-->what do you mean with tools only ? wireguard nothing else ?  No, problem for me ;)

hope you can help me out 

thanks a lot !!!!

**Comment by bmullan** [comment]
@gamer123  When you have time could you document what you did to install/configure Wireguard on 20.04 for the Raspberry Pi and post it to either the **[sub-reddit for LXC](https://www.reddit.com/r/LXC/)** or the **[sub-reddit for LXD](https://www.reddit.com/r/LXD/)** as appropriate (ie LXC or LXD)

fyi I am the moderator of both & think there would be alot of interest in how to do this

**Comment by gamer123** [comment]
The configuration on the Raspberry is not running in a LXC or LXD container it is running directly in the OS. 

I try to set up the config on an QNAP in a LXC container but i have trouble with that because the interface wg0 can`t start. 

If i get a anwser i can do this it is not that much you have to do.  In my case the tunnel works in mode IPv6 only. So it is IPv4/v6 in IPv6 only. That means both devices must have a IPv6 for me it is mobil telekom and mobil vodafone they provide Dualstack. The mobil O2 guys have no chance to use this at the moment (09.09.2020) ..  The guys with a new contract from Vodafone, Telekom or O2 wired Connection (DSL or cable) should have a IPv6.

**Comment by bmullan** [comment]
You might also ask this question on the LXD/LXC forum & see if the devs etc
have an idea for you

https://discuss.linuxcontainers.org/

**Comment by Haxprox** [comment]
Just for those who are stuck here, check this old article: https://nixvsevil.com/posts/wireguard-in-proxmox-lxc
And delete this function calling for using this script: https://github.com/angristan/wireguard-install/blob/574cc06b31c73cb3b57ffb51b0b025052be69886/wireguard-install.sh#L60
Enjoy!

**Comment by aHVzY2g** [comment]
@Haxprox thx for the hint.

Additionally, I had to run `dpkg-reconfigure wireguard-dkms` on the host and recreate the container.
